src/components/Deck.js

```diff
@@ -49,18 +48,30 @@ const Deck = () => {
     }
   }
 
+  const reset = () => {
+    trickDeck = randomizeArray(deck).slice(0, TOTAL_OF_CARDS);
```

Comment:
Em vez de reatribuir uma variável fora do componente (tornando o componente menos reutilizável), o que acha de atribuir o novo valor a uma variável local? Algo assim:

```suggestion
    const newTrickDeck = randomizeArray(deck).slice(0, TOTAL_OF_CARDS);
```

E aí usar esta nova variável pra criar as colunas. Uma sugestão junto com essa seria deixar a lógica de criar um deck e gerar as colunas isolada numa função. Algo como:

```js
const columns = createColumns(deck)
```

E dentro do `createColumns` já faríamos tanto a randomização, o slice, e a separação por colunas. Que acha?